src/main/java/com/example/demo/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+    private final BookRepository bookRepository;
+
+    @Override
+    public ShoppingCartDto findShoppingCart(Authentication authentication, Pageable pageable) {
+        User user = (User) authentication.getPrincipal();
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException("Can't find ShoppingCart by userId: "
+                        + user.getId() + ".Put some items to ShoppingCart to create it."));
+
+        ShoppingCartDto shoppingCartDto = shoppingCartMapper.toDto(shoppingCart);
+        Set<CartItemResponseDto> cartItemsSet = cartItemRepository
+                .findByShoppingCartId(shoppingCart.getId(), pageable)
+                .stream()
+                .map(cartItemMapper::toDto)
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+        shoppingCartDto.setCartItems(cartItemsSet);
+        return shoppingCartDto;
+    }
+
+    @Override
+    public CartItemResponseDto addBookToShoppingCart(Authentication authentication,
+                                                     CartItemRequestDto requestDto) {
+        User user = (User) authentication.getPrincipal();
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId())
+                .orElseGet(() -> createNewShoppingCart(user));
+
+        Book book = bookRepository.findById(requestDto.getBookId())
+                .orElseThrow(() -> new EntityNotFoundException("Book by current id: "
+                        + requestDto.getBookId() + " doesn't exist"));
+
+        CartItem cartItem = cartItemMapper.toModel(requestDto);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItem.setBook(book);
+        CartItem savedCartItem = cartItemRepository.save(cartItem);
+        Set<CartItem> cartItems = cartItemRepository.findByShoppingCartId(shoppingCart.getId());
+        cartItems.add(savedCartItem);
+        shoppingCart.setCartItems(cartItems);
+        shoppingCartRepository.save(shoppingCart);
+        return cartItemMapper.toDto(cartItem);
+    }
+
+    @Override
+    public CartItemResponseDto updateBookQuantityInTheShoppingCart(Authentication authentication,
+                                                         Long cartItemId,
+                                                         CartItemUpdateDataRequestDto requestDto) {
+        User user = (User) authentication.getPrincipal();
+        CartItem cartItem = getValidCartItem(cartItemId);
+        ShoppingCart shoppingCart = getValidShoppingCart(user.getId());
+        if (cartItemRepository.findByShoppingCartId(shoppingCart.getId()).contains(cartItem)) {
+            cartItem.setQuantity(requestDto.getQuantity());
+            cartItemRepository.save(cartItem);
+        } else {
+            throw new EntityNotFoundException("Can't find CartItem by id: " + cartItemId
+                    + " in the ShoppingCart to update");
+        }
```

Comment:
it's better to throw exception early (invert condition) and unwrap the rest of your code to make it more readable